### ISSUE TITLE
Update SA1200 to treat extern alias directives as using directives

### ISSUE THREAD

**Issue by sharwell** [DotNetAnalyzers/StyleCopAnalyzers#1931]
Currently both SA1200 and `UsingCodeFixProvider` ignore extern alias directives. I propose that these directives be treated as using directives for the purposes of both reporting violations and fixing the code.

**Comment by sharwell** [comment]
@vweijsters Can you check and see what StyleCop Classic did in this case? If our current behavior matches StyleCop Classic, then this is an enhancement proposal. Otherwise, this would be a bug that we can just fix for the next release.

**Comment by vweijsters** [comment]
StyleCop classic ignores `extern alias`.

I assume that you mean that `extern alias` should be sorted on identifier? Because `using` and `extern alias` elements cannot be mixed. Given the different nature of both elements, I would propose to create a new rule for sorting `extern alias` on identifier.

**Comment by sharwell** [comment]
@vweijsters No, I'm referring to the placement of extern alias directives inside or outside of namespace declarations.

**Comment by vweijsters** [comment]
That makes sense, I confused SA1200 with SA1210.
I would not be very happy to add it to the `UsingCodeFixProvider` (as it is already very complex), but it might be unavoidable, as using statements that are using the external aliases might be moved outside by the codefix, potentially breaking stuff.
